OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -899,255 +1152,574 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
+            DreamValue second = state.Pop();
+            DreamValue first = state.Pop();
+
+            if(first == DreamValue.Null) {
+                state.Push(new DreamValue(0.0f));
+                return null; //early return for null / anything = 0
+            }
+
+            switch(first.Type) {
+                case DreamValue.DreamValueType.Float: {
+                    if (first.TryGetValueAsFloat(out var firstFloat) && second.TryGetValueAsFloat(out var secondFloat)) {
+                        if (secondFloat == 0)
+                            throw new Exception("Division by zero");
+
+                        state.Push(new DreamValue(firstFloat / secondFloat));
+                        return null;
+                    }
+                    break;
+                }
+                case DreamValue.DreamValueType.String: {
+                    break;
+                }
+                case DreamValue.DreamValueType.DreamObject: {
+                    if(first.TryGetValueAsDreamObject(out DreamObject? obj)) {
+                        IDreamMetaObject? metaObject = obj?.ObjectDefinition?.MetaObject;
+                        return metaObject?.OperatorDivide(first, second, state);
+                    }
+                    break;
+                }
+                case DreamValue.DreamValueType.DreamProc:
+                case DreamValue.DreamValueType.DreamResource:
+                default:
+                    break;
+            }
+            //no condition exists to handle the inputs, so error
+            throw new InvalidOperationException($"Divide cannot be done between {first} and {second}");
+        }
+
+        public static ProcStatus? DivideReference(DMProcState state) {
             DMReference reference = state.ReadReference();
             DreamValue second = state.Pop();
             DreamValue first = state.GetReferenceValue(reference, peek: true);
 
-            DreamValue result;
-            if (first.TryGetValueAsDreamObject(out var firstObj)) {
-                if (firstObj != null) {
-                    IDreamMetaObject metaObject = firstObj.ObjectDefinition.MetaObject;
+            DreamValue output;
+
+            if(first == DreamValue.Null) {
+                output = new DreamValue(0);
+                state.AssignReference(reference, output);
+                state.Push(output);
+                return null; //early return for null / anything = 0
+            }
+
+            switch(first.Type) {
+                case DreamValue.DreamValueType.Float: {
+                     if (first.TryGetValueAsFloat(out var firstFloat) && second.TryGetValueAsFloat(out var secondFloat)) {
+                        if (secondFloat == 0)
+                            throw new Exception("Division by zero");
+
+                        output = new DreamValue(firstFloat / secondFloat);
+                        state.AssignReference(reference, output);
+                        state.Push(output);
```

Comment:
There's no `return null;` to end this and prevent the invalid-divide-ref exception.
```suggestion
                        state.Push(output);
                        return null;
```